pallets/check-membership/src/loose/mod.rs

```diff
@@ -0,0 +1,64 @@
+//! Pallet that demonstrates a minimal access control check. When a user calls this pallet's
+//! only dispatchable function, `check_membership`, the caller is checked against a set of approved
+//! callers. Only if the caller is approved, do they successfully emit the event.
+//!
+//! The list of approved members is provided by an external source and referenced via an associated
+//! type in this pallet's configuration trait. Any type that implements the AccountSet trait can be
+//! used to supply the membership set.
+
+use frame_support::{decl_error, decl_event, decl_module, dispatch::DispatchResult, ensure};
+use frame_system::{self as system, ensure_signed};
+use account_set::AccountSet;
+
+#[cfg(test)]
+mod tests;
+
+/// The pallet's configuration trait
+/// Notice the loose coupling to any pallet that implements the AccountSet behavior
+pub trait Trait: system::Trait {
+	/// The ubiquitous event type
+	type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+
+	/// A type that will supply a set of members to check access control against
+	type MembershipSource: AccountSet<AccountId = Self::AccountId>;
+}
+
+decl_event!(
+	pub enum Event<T>
+	where
+		AccountId = <T as system::Trait>::AccountId,
+	{
+		/// The caller is a member.
+		IsAMember(AccountId),
+	}
+);
+
+decl_error! {
+	pub enum Error for Module<T: Trait> {
+		/// The caller is not a member
+		NotAMember,
+	}
+}
+
+decl_module! {
+	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+		fn deposit_event() = default;
+
+		/// Checks whether the caller is a member of the set of Account Ids provided by the
+		/// MembershipSource type. Emits an event if they are, and errors if not.
+		#[weight = 10_000]
+		fn check_membership(origin) -> DispatchResult {
+			let caller = ensure_signed(origin)?;
+
+			// Get the members from the vec-set pallet
```

Comment:
Do you really mean to use `vec-set` here?
```suggestion
			// Get the members from the `vec-set` pallet
```